modules/vaos/app/services/vaos/v2/appointments_service.rb

```diff
@@ -273,10 +292,20 @@ def merge_appointments(eps_appointments, appointments)
         normalized_new = eps_appointments.map(&:serializable_hash)
         existing_referral_ids = appointments.to_set { |a| a.dig(:referral, :referral_number) }
         date_and_time_for_referral_list = appointments.pluck(:start)
+
+        # Track which EPS appointments get rejected as duplicates
+        rejected_ids = []
         merged_data = appointments + normalized_new.reject do |a|
-          existing_referral_ids.include?(a.dig(:referral,
-                                               :referral_number)) && date_and_time_for_referral_list.include?(a[:start])
+          duplicate = existing_referral_ids.include?(a.dig(:referral, :referral_number)) &&
+                      date_and_time_for_referral_list.include?(a[:start])
+          rejected_ids << a[:id] if duplicate
+          duplicate
         end
+
+        kept_eps_ids = normalized_new.pluck(:id) - rejected_ids
+        duplicates_msg = rejected_ids.any? ? ", removed duplicates #{rejected_ids}" : ''
+        Rails.logger.info("EPS Debug: Merge kept #{kept_eps_ids}#{duplicates_msg}")
```

Comment:
I think this needs to be guarded by the feature flipper check, otherwise this will always log appointment ids